modules/nw-ingress-setting-update-subnet-LoadBalancerService.adoc

```diff
@@ -0,0 +1,80 @@
+
+.Procedure
+To update an `IngressController` with manually specified load balancer subnets, you can follow these steps:
+
+. Choose the subnets.
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name:  <name> <1>
+  namespace: openshift-ingress-operator
+spec:
+  domain: <domain> <2>
+  endpointPublishingStrategy:
+    type: LoadBalancerService
+    loadBalancer:
+      scope: External
+      providerParameters:
+        type: AWS
+        aws:
+          type: Classic <3>
+          classicLoadBalancer: <4>
+            subnets:
+              ids: <5>
+              - <updated_subnet> <6>
+              - <updated_subnet>
+              - <updated_subnet>
+----
+<1> Replace `<name>` with a name for the `IngressController`.
+<2> Replace `<domain>` with the DNS name serviced by the `IngressController`.
+<3> Specify updated subnet IDs (or names if you using `names`).
+<4> You can also use the `networkLoadBalancer` field if using an NLB.
+<5> You can optionally specify a subnet by name using the `names` field instead of specifying the subnet by ID.
+<6> Specify subnet IDs (or names if you using `names`).
+
+. Examine the `Progressing` condition on the `IngressController` for instructions on how to apply the subnet updates:
++
+[source,terminal]
+----
+$ oc get ingresscontroller -n openshift-ingress-operator subnets -o jsonpath="{.status.conditions[?(@.type==\"Progressing\")]}" | yq -PC
+----
++
+.Example output
+[source,terminal]
+----
+lastTransitionTime: "2024-11-25T20:19:31Z"
+message: 'One or more status conditions indicate progressing: LoadBalancerProgressing=True (OperandsProgressing: One or more managed resources are progressing: The IngressController subnets were changed from [...] to [...].  To effectuate this change, you must delete the service: `oc -n openshift-ingress delete svc/router-<name>`; the service load-balancer will then be deprovisioned and a new one created. This will most likely cause the new load-balancer to have a different host name and IP address and cause disruption. To return to the previous state, you can revert the change to the IngressController: [...]'
+reason: IngressControllerProgressing
+status: "True"
+type: Progressing
+----
+
+. Delete the service associated with the `IngressController` to apply the update:
++
+[source,terminal]
+----
+$ oc -n openshift-ingress delete svc/router-<name>
+----
++
+[WARNING]
+====
+An outage that can last several minutes due to new DNS records propagation, new load balancers provisioning, and other factors. IP addresses and canonical names of the Ingress Controller load balancer might change after applying this procedure.
```

Comment:
First sentence seems to be missing a noun or something, not sure what is needed to fix it.
```suggestion
An outage that can last several minutes due to new DNS records propagation, new load balancers provisioning, and other factors. IP addresses and canonical names of the Ingress controller load balancer might change after applying this procedure.
```

Reply:
Agreed. I think the problem is the "that" that shouldn't be there.